gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -656,20 +656,16 @@ def uri_escape_path(string)
         self.class.uri_escape_path(string)
       end
 
-      def get_credentials
+      def fetch_credentials
         credentials = @credentials_provider.credentials
-        if credentials_set?(credentials)
+        if credentials.access_key_id && credentials.secret_access_key
```

Comment:
I'm finding it impossible with the previous version as well as this version to not produce the deprecation warning. Is this intentional? It seems as best I can tell this code is simultaneously deprecating `#access_key_id` and `#secret_access_key` as well as still using the methods internally.

Reply:
In section of code, I removed the credentials_set? method as it was only called once in this location. There are no deprecations going on in the sigv4 signer - this was a small refactor that I included into the change.

We are only deprecating those methods for credential provider objects, not the provider itself. On line 660, we're fetching credentials via the provider and then calling those methods on credentials.